extensions/fetch/Cargo.toml

```diff
@@ -13,13 +13,14 @@ repository = "https://github.com/denoland/deno"
 [lib]
 path = "lib.rs"
 
+
 [dependencies]
 bytes = "1.0.1"
 data-url = "0.1.0"
 deno_core = { version = "0.94.0", path = "../../core" }
 deno_web = { version = "0.43.0", path = "../web" }
 http = "0.2.4"
-reqwest = { version = "0.11.4", default-features = false, features = ["rustls-tls", "stream", "gzip", "brotli"] }
+reqwest = { version = "0.11.4", default-features = false, features = ["rustls-tls", "native-tls", "stream", "gzip", "brotli"] }
```

Comment:
Unfortunately this is a no-go - using `native-tls` feature in `reqwest` will link to OpenSSL, which we explicitly gave up on doing last year.

Reply:
So I'm looking at using `load_native_certs()` and then trying to pass those to the ClientBuilder instead of using its implementation. I will remove the `native-tls` feature.

I'm hitting a small roadblock, `load_native_certs()` returns a set of `OwnedTrustAnchor`'s and the `ClientBuilder` doesn't expose a way to access the root_store or add TrustAnchor's, unless someone better rust-fu can understand how to convert a TrustAnchor into a Certificate, it seems like it will require a small change to reqwest to enable it.

Here it is adding the same webpki certs in the same way deno is:
https://github.com/seanmonstar/reqwest/blob/bccefe7486f32b9289a3d7fe5790e74cd68a879a/src/async_impl/client.rs#L311

So I think probably updating the ClientBuilder api to have something like `add_trust_anchor` function would enable us to do this:

```ts
load_native_certs()
  .expect("could not load platform certs")
  .roots
  .map(|root: OwnedTrustAnchor| builder.add_trust_anchor(root));
```

Reply:
> unless someone better rust-fu can understand how to convert a TrustAnchor into a Certificate, it seems like it will require a small change to reqwest to enable it.

I don't think that's possible, I guess a change to reqwest is needed after all

Reply:
You can use `ClientBuilder::use_preconfigured_tls`

Reply:
Yup that worked. Though it required me to copy the implementation for `CLIENT_SESSION_MEMORY_CACHE` which appeaers to only exist in the `net` extension right now. Can I reference the net module here and access that static variable so its shared or should it be bumped up to a different module where they can both access it? Or just duplicating it in each module is fine?